server/webhook.go

```diff
@@ -123,11 +172,213 @@ func (p *Plugin) handleMeetingEnded(w http.ResponseWriter, r *http.Request, body
 		return
 	}
 
-	if appErr = p.deleteMeetingPostID(meetingPostID); appErr != nil {
-		p.API.LogWarn("failed to delete db entry", "error", appErr.Error())
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(post); err != nil {
+		p.API.LogWarn("failed to write response", "error", err.Error())
+	}
+}
+
+func (p *Plugin) handleTranscript(recording zoom.RecordingFile, postID, channelID, downloadToken string) error {
+	request, err := http.NewRequest(http.MethodGet, recording.DownloadURL, nil)
+	if err != nil {
+		p.API.LogWarn("Unable to get the transcription", "err", err)
+		return err
+	}
+	request.Header.Set("Authorization", "Bearer "+downloadToken)
+
+	retries := 5
+	var response *http.Response
+	for retries > 0 {
+		response, err = http.DefaultClient.Do(request)
+		if err != nil {
+			p.API.LogWarn("Unable to get the transcription", "err", err)
+			time.Sleep(1 * time.Second)
+			retries--
+			continue
+		}
+
+		if response.StatusCode != http.StatusOK {
+			response.Body.Close()
+			p.API.LogWarn("Unable to get the transcription", "err", "bad status code "+strconv.Itoa(response.StatusCode))
+			time.Sleep(1 * time.Second)
+			retries--
+			continue
+		}
+		break
+	}
+
+	if response == nil {
+		p.API.LogWarn("Unable to get the transcription", "err", "response is nil")
+		return err
+	}
+
+	defer response.Body.Close()
+	transcription, err := io.ReadAll(response.Body)
+	if err != nil {
+		p.API.LogWarn("Unable to get the transcription", "err", err)
+		return err
+	}
+	fileInfo, appErr := p.API.UploadFile(transcription, channelID, "transcription.txt")
+	if appErr != nil {
+		p.API.LogWarn("Unable to get the transcription", "err", appErr)
+		return appErr
+	}
+	newPost := &model.Post{
+		UserId:    p.botUserID,
+		ChannelId: channelID,
+		RootId:    postID,
+		Message:   "Here's the zoom meeting transcription",
+		FileIds:   []string{},
+		Type:      "custom_zoom_transcript",
+	}
+
+	newPost.FileIds = append(newPost.FileIds, fileInfo.Id)
+	newPost.AddProp("captions", []any{map[string]any{"file_id": fileInfo.Id}})
+
+	_, appErr = p.API.CreatePost(newPost)
+	if appErr != nil {
+		p.API.LogWarn("Could not update the post", "err", appErr)
+		return appErr
+	}
+
+	return nil
+}
+
+func (p *Plugin) handleTranscriptCompleted(w http.ResponseWriter, r *http.Request, body []byte) {
+	var webhook zoom.RecordingWebhook
+	if err := json.Unmarshal(body, &webhook); err != nil {
+		p.API.LogError("Error unmarshaling meeting webhook", "err", err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	meetingPostID := webhook.Payload.Object.UUID
+	postID, appErr := p.fetchMeetingPostID(meetingPostID)
+	if appErr != nil {
+		http.Error(w, appErr.Error(), appErr.StatusCode)
 		return
 	}
 
+	post, appErr := p.API.GetPost(postID)
```

Comment:
1/5 We should try to use `p.client` instead of `p.API` for new code added

Reply:
I would suggest to do that as one shot, in another PR, It is consistent with the whole codebase and I prefer to don't change it here.